File: pruebaAppBar/app/src/main/java/com/p2/pruebaappbar/RecyclerViewPrueba.java
package com.p2.pruebaappbar;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewpager.widget.ViewPager;
import androidx.viewpager2.widget.ViewPager2;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.google.android.material.tabs.TabLayout;

import java.util.ArrayList;
import java.util.Random;

public class RecyclerViewPrueba extends AppCompatActivity {
    Toolbar toolbar;
    ArrayList<Disco> listaDiscos = new ArrayList<>();
    TabLayout misTabs;
    ViewPager viewPager;
    FragmentPagerAdapter adaptadorFragmento;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        for (int i = 0; i < 2; i++) {
            listaDiscos.add(new Disco("TGG", "The Wonder Years", R.drawable.tgg));
            listaDiscos.add(new Disco("Suburbia", "The Wonder Years", R.drawable.suburbia));
            listaDiscos.add(new Disco("Sister cities", "The Wonder Years", R.drawable.sc));
            listaDiscos.add(new Disco("NCTH", "The Wonder Years", R.drawable.ncth));
        }

        toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        viewPager = findViewById(R.id.viewPager);
        adaptadorFragmento = new FragmentPagerAdapter(getSupportFragmentManager(),FragmentPagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
        adaptadorFragmento.setListaDiscos(listaDiscos);
        viewPager.setAdapter(adaptadorFragmento);
        misTabs = findViewById(R.id.tabs);
        misTabs.setTabMode(TabLayout.MODE_SCROLLABLE);
        misTabs.setupWithViewPager(viewPager);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.mimenu,menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

        switch (item.getItemId()) {
            case R.id.crear:
                listaDiscos.add(0,new Disco("Disco Creado","autor inventado",R.drawable.tgg));
                adaptadorFragmento.setListaDiscos(listaDiscos);
                adaptadorFragmento.notifyDataSetChanged();
                return true;

            case R.id.modificar:
                int posicionAzar = new Random().nextInt(listaDiscos.size());
                listaDiscos.set(posicionAzar,new Disco("Disco CAMBIADO!!","autor CAMBIADO!!",R.drawable.ncth));
                //adaptadorDisco.notifyItemChanged(posicionAzar);
                adaptadorFragmento.setListaDiscos(listaDiscos);
                adaptadorFragmento.notifyDataSetChanged();
                return true;
            case R.id.borrar:
                if(listaDiscos.size() > 1) {
                    listaDiscos.remove(0);
                    //adaptadorDisco.notifyItemRemoved(0);
                    adaptadorFragmento.setListaDiscos(listaDiscos);
                    adaptadorFragmento.notifyDataSetChanged();
                }
                return true;
            default:
                return super.onOptionsItemSelected(item);

        }
    }

}

File: pruebaAppBar/app/src/main/java/com/p2/pruebaappbar/RecyclerViewPrueba.kt
package com.p2.pruebaappbar


import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.google.android.material.tabs.TabLayout
import androidx.viewpager.widget.ViewPager
import androidx.fragment.app.FragmentStatePagerAdapter
import java.util.*

class RecyclerViewPrueba : AppCompatActivity() {
    lateinit  var toolbar: Toolbar
    var listaDiscos = ArrayList<Disco>()
    lateinit var misTabs: TabLayout
    lateinit var viewPager: ViewPager
    lateinit var adaptadorFragmento: FragmentPagerAdapter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        for (i in 0..1) {
            listaDiscos.add(Disco("TGG", "The Wonder Years", R.drawable.tgg))
            listaDiscos.add(Disco("Suburbia", "The Wonder Years", R.drawable.suburbia))
            listaDiscos.add(Disco("Sister cities", "The Wonder Years", R.drawable.sc))
            listaDiscos.add(Disco("NCTH", "The Wonder Years", R.drawable.ncth))
        }
        toolbar = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)
        viewPager = findViewById(R.id.viewPager)
        adaptadorFragmento = FragmentPagerAdapter(supportFragmentManager, FragmentStatePagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT)
        adaptadorFragmento.listaDiscos = listaDiscos
        viewPager.setAdapter(adaptadorFragmento)
        misTabs = findViewById(R.id.tabs)
        misTabs.setTabMode(TabLayout.MODE_SCROLLABLE)
        misTabs.setupWithViewPager(viewPager)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.mimenu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.crear -> {
                listaDiscos.add(0, Disco("Disco Creado", "autor inventado", R.drawable.tgg))
                adaptadorFragmento.notifyDataSetChanged()
                true
            }
            R.id.modificar -> {
                val posicionAzar = Random().nextInt(listaDiscos.size)
                listaDiscos[posicionAzar] = Disco("Disco CAMBIADO!!", "autor CAMBIADO!!", R.drawable.ncth)
                //adaptadorDisco.notifyItemChanged(posicionAzar);
                adaptadorFragmento.notifyDataSetChanged()
                true
            }
            R.id.borrar -> {
                if (listaDiscos.size > 1) {
                    listaDiscos.removeAt(0)
                    //adaptadorDisco.notifyItemRemoved(0);
                    adaptadorFragmento.notifyDataSetChanged()
                }
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}